allspice/utils/bom_generation.py

```diff
@@ -150,54 +150,44 @@ def generate_bom(
         a column name and the value is the value for that column.
     """
 
-    if source_file.lower().endswith(".prjpcb"):
-        project_tool = SupportedTool.ALTIUM
-    elif source_file.lower().endswith(".dsn"):
-        project_tool = SupportedTool.ORCAD
-    elif source_file.lower().endswith(".sdax"):
-        project_tool = SupportedTool.SYSTEM_CAPTURE
-    else:
-        raise ValueError(
-            "The source file must be a PrjPcb file for Altium projects or a DSN file for OrCAD "
-            "projects."
+    allspice_client.logger.info(
+        f"Generating BOM for {repository.get_full_name()=} on {ref=} using {columns=}"
+    )
+
+    if group_by is not None:
+        for group_column in group_by:
+            if group_column not in columns:
+                raise ValueError(f"Group by column {group_column} not found in selected columns")
+
+    components = list_components(
+        allspice_client,
+        repository,
+        source_file,
+        variant,
+        ref,
+        combine_multi_part=True,
+    )
+
+    if remove_non_bom_components:
+        project_tool = infer_project_tool(source_file)
+        if project_tool == SupportedTool.ALTIUM:
+            components = _remove_non_bom_components(components)
```

Comment:
Unless the plan is to use the same function for the other file formats I think we should prefix this as altium specific.